src/lab13.py

```diff
@@ -16,8 +16,39 @@
 import time
 import urllib.parse
 from lab4 import print_tree
-from lab4 import Element
-from lab4 import Text
+
+class Text:
+    def __init__(self, text, parent):
+        self.text = text
+        self.children = []
+        self.parent = parent
+
+        self.style = {}
+        self.animations = {}
+
+    def __repr__(self):
+        return repr(self.text)
+
+class Element:
+    def __init__(self, tag, attributes, parent):
+        self.tag = tag
+        self.attributes = attributes
+        self.children = []
+        self.parent = parent
+
+        self.style = {}
+        self.animations = {}
+
+    def __repr__(self):
+        attrs = [" " + k + "=\"" + v + "\"" for k, v  in self.attributes.items()]
+        return "<" + self.tag + "".join(attrs) + ">"
+
+# Patch the `Text` and `Element` classes so that all other code that
+# uses them, like HTMLParser, all refer to the patched versions.
+import sys
+sys.modules['lab4'].Text = Text
```

Comment:
Interesting. we could do this to a bunch of other chapters to avoid redundant code.